src/org/traccar/model/Position.java

```diff
@@ -22,9 +22,11 @@
     public static final String KEY_ORIGINAL = "raw";
     public static final String KEY_INDEX = "index";
     public static final String KEY_HDOP = "hdop";
-    public static final String KEY_SATELLITES = "sat";
+    public static final String KEY_SATELLITES = "sat"; // in use
+    public static final String KEY_SATELLITES_VISIBLE = "satVisible";
     public static final String KEY_RSSI = "rssi";
     public static final String KEY_GPS = "gps";
+    public static final String KEY_FIX = "fix"; // GPS, GSM, none
```

Comment:
This should probably be hardcoded in the protocol as well. Most likely you need to map this to position.valid flag.